service/implementation/email_sender/user_subscription_manager.py

```diff
@@ -0,0 +1,99 @@
+import os
+import base64
+
+from dependency_injector.wiring import Provide, inject
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+from googleapiclient.discovery import build
+from email.mime.text import MIMEText
+
+from database.models import TempSubscribersData
+
+from pathlib import Path
+from dotenv import load_dotenv
+
+from sqlalchemy import event
+
+from api.container.container import Container
+from database.postgres.dal.user_subscription import UserSubscriptionDAL
+
+
+class UserSubscriptionManagerMeta(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            _instance = super().__call__(*args, **kwargs)
+            cls._instances[cls] = _instance
+        return cls._instances[cls]
+
+class UserSubscriptionManager(metaclass=UserSubscriptionManagerMeta):
+    __SCOPES = ["https://www.googleapis.com/auth/gmail.send"]
+
+    @inject
+    def __init__(self, user_subscription_dal: UserSubscriptionDAL = Provide[Container.user_subscription_dal]):
+        self.__user_subscription_dal = user_subscription_dal
+
+        current_dir = Path(__file__).resolve().parent if "__file__" in locals() else Path.cwd()
+        envars = current_dir / ".env"
+        load_dotenv(envars)
+
+        self.__sender_email = os.getenv("EMAIL")
+        self.__sender_password = os.getenv("PASSWORD")
+
+        event.listen(TempSubscribersData, "after_insert", self.__on_subscribers_inserted)
+
+    @inject
+    def try_add_subscribers_to_temp_table(self, team_index):
+        self.__user_subscription_dal.try_add_subscribers_data(team_index)
+
+    def __on_subscribers_inserted(self, mapper, connection, target):
+        self.__try_send_email_to_users()
+
+    def __try_send_email_to_users(self):
+        subscribed_users = self.__user_subscription_dal.get_subscribed_users_data_and_delete_rows()
```

Comment:
If you have to write word "AND" in the name of your function: that's like 99.9% chance your method is not SOLID.

Single responsibility, pls